implementations/rust/ockam/ockam/src/entity/local.rs

```diff
@@ -0,0 +1,219 @@
+use crate::{
+    Contact, ContactsDb, Entity, KeyAttributes, Profile, ProfileAuth, ProfileChangeEvent,
+    ProfileChanges, ProfileContacts, ProfileEventAttributes, ProfileIdentifier, ProfileIdentity,
+    ProfileSecrets, ProfileSync, RemoteEntity, RemoteForwarder, RemoteForwarderInfo, Route,
+    SecureChannelTrait,
+};
+use ockam_core::{Address, AddressSet, Message, Result, Worker};
+use ockam_node::{Cancel, Context};
+use ockam_vault_core::{PublicKey, Secret};
+use ockam_vault_sync_core::Vault;
+
+use ockam_core::hashbrown::HashMap;
+
+pub struct LocalEntity {
+    pub ctx: Context,
+    vault: Address,
+    entity: Entity<ProfileSync>,
+    secure_channels: HashMap<Address, Route>,
```

Comment:
It would be great to add some additional data to each channel in the future. And I'm not sure Route that is stored here is useful

Reply:
This is pretty much only here for the proof of concept `list_secure_channels` API that we mentioned in the design call. Mrinal do you think this is useful at this point? Should we create some higher level type that encapsulates Route/whatever metadata?